cmd/clusterctl/client/repository/components_test.go

```diff
@@ -105,21 +108,20 @@ func Test_replaceVariables(t *testing.T) {
 	for _, tt := range tests {
 		t.Run(tt.name, func(t *testing.T) {
 			got, err := replaceVariables(tt.args.yaml, tt.args.variables, tt.args.configVariablesClient)
-			if (err != nil) != tt.wantErr {
-				t.Fatalf("error = %v, wantErr %v", err, tt.wantErr)
-			}
 			if tt.wantErr {
+				g.Expect(err).To(HaveOccurred())
 				return
 			}
 
-			if !reflect.DeepEqual(got, tt.want) {
-				t.Errorf("got = %v, want %v", got, tt.want)
-			}
+			g.Expect(err).NotTo(HaveOccurred())
```

Comment:
nit we should be consistent in having assertion on the operation err/on wantErr  like in Test_componentsClient_Get
```
if tt.wantErr {
   g.Expect(err).To(HaveOccurred())
   return
}				
g.Expect(err).NotTo(HaveOccurred())

```
Without empty line in the middle and with an empty line after